src/components/datePicker/header/datePickerHeader.tsx

```diff
@@ -0,0 +1,123 @@
+import { FC, ReactNode, useMemo } from 'react';
+import classNames from 'classnames/bind';
+import { Dropdown } from '@/components/dropdown';
+import { CalendarArrowIcon } from '@components/icons';
+import { DropdownOptionType, DropdownValue } from '@components/dropdown/types';
+import styles from './datePickerHeader.module.scss';
+
+const cx = classNames.bind(styles);
+
+const getYearsFrom = (start: number, amountYearsToGenerate = 20) => {
+  const yearsFromCurrent = start + amountYearsToGenerate;
+  return new Array(yearsFromCurrent - start).fill(undefined).map((_, i) => start - i);
+};
+
+export interface DatePickerHeaderProps {
+  changeYear: (year: number) => void;
+  changeMonth: (month: number) => void;
+  decreaseMonth: () => void;
+  increaseMonth: () => void;
+  headerNodes: ReactNode;
+  date: Date;
+  prevMonthButtonDisabled: boolean;
+  nextMonthButtonDisabled: boolean;
+  customClassName: string;
+  yearsOptions: number[];
+  locale: string;
+}
+
+export const DatePickerHeader: FC<DatePickerHeaderProps> = ({
+  date = new Date(),
+  changeYear = () => {},
+  changeMonth = () => {},
+  decreaseMonth = () => {},
+  increaseMonth = () => {},
+  prevMonthButtonDisabled = false,
+  nextMonthButtonDisabled = false,
+  headerNodes = null,
+  customClassName = '',
+  yearsOptions = [],
+  locale,
+}) => {
+  const year = date.getFullYear();
+  const month = date.getMonth();
+
+  const monthList = Array(12).keys();
+  const formatter = new Intl.DateTimeFormat(locale, {
+    month: 'long',
+  });
+  const getMonthName = (monthIndex: number) => formatter.format(new Date(year, monthIndex));
+  const months = Array.from(monthList, getMonthName);
```

Comment:
Why not to create this variable within `monthDropdownOptions` creation?
It is used only there.